stake-pool/single-pool-cli/src/quarantine.rs

```diff
@@ -0,0 +1,197 @@
+use {
+    base64::{prelude::BASE64_STANDARD, Engine},
+    solana_sdk::{
+        instruction::Instruction,
+        native_token::LAMPORTS_PER_SOL,
+        pubkey::Pubkey,
+        signature::{Keypair, Signer},
+        stake::{
+            self,
+            state::{Authorized, Lockup, Meta, Stake, StakeState},
+        },
+        system_instruction, system_program,
+        sysvar::{self, rent::Rent},
+        transaction::Transaction,
+    },
+    solana_vote_program::{
+        vote_instruction::{self, CreateVoteAccountConfig},
+        vote_state::{VoteInit, VoteState, VoteStateVersions},
+    },
+};
+
+use crate::config::*;
+
+// XXX this file is stuff that should not be in the cli
+// once i have a better understanding of needs/affordances im making a stake program client
+
+pub async fn get_rent(config: &Config) -> Result<Rent, Error> {
+    let rent_data = config
+        .program_client
+        .get_account(sysvar::rent::id())
+        .await?
+        .unwrap();
+    let rent = bincode::deserialize::<Rent>(&rent_data.data)?;
+
+    Ok(rent)
+}
+
+pub async fn get_minimum_delegation(config: &Config) -> Result<u64, Error> {
+    let payer = config.fee_payer()?;
+
+    let transaction = Transaction::new_signed_with_payer(
+        &[stake::instruction::get_minimum_delegation()],
+        Some(&payer.pubkey()),
+        &vec![payer],
+        config.program_client.get_latest_blockhash().await?,
+    );
+    let return_data = config
+        .rpc_client
+        .simulate_transaction(&transaction)
+        .await?
+        .value
+        .return_data
+        .unwrap()
+        .data;
+    let mut data = BASE64_STANDARD.decode(return_data.0)?;
+    data.resize(8, 0);
+    Ok(data.try_into().map(u64::from_le_bytes).unwrap())
+}
+
+pub async fn get_stake_info(
+    config: &Config,
+    stake_account_address: &Pubkey,
+) -> Result<Option<(Meta, Stake)>, Error> {
+    if let Some(stake_account) = config
+        .program_client
+        .get_account(*stake_account_address)
+        .await?
+    {
+        match bincode::deserialize::<StakeState>(&stake_account.data)? {
+            StakeState::Stake(meta, stake) => Ok(Some((meta, stake))),
+            StakeState::Initialized(_) => {
+                Err(format!("Stake account {} is undelegated", stake_account_address).into())
+            }
+            _ => unimplemented!(),
```

Comment:
Just in case there are zombie accounts, it'd be good to also handle `StakeState::Uninitialized`